### PR TITLE
Fix #1397 Add missing properties in conversation.list API response

### LINKED ISSUE #1397: Some fields of the type definition of conversations.list are missing.
#### Packages:

Select all that apply:

- [x] `@slack/web-api`
- [ ] `@slack/rtm-api`
- [ ] `@slack/webhooks`
- [ ] `@slack/oauth`
- [ ] `@slack/socket-mode`
- [ ] I don't know

I noticed that the type definition of `conversations.list API` is partly missing.
This may be a problem with java-slack-sdk, but I will write it here.

In [ConversationsListResponse.ts](https://github.com/slackapi/node-slack-sdk/blob/main/packages/web-api/src/response/ConversationsListResponse.ts), this may not be all, but some fields are missing, like `user`, `is_user_deleted`, and `priority`. As far as I can see from the [docs](https://api.slack.com/methods/conversations.list), it seems that if `is_im` is `true`, i.e. in the case of direct message, these fields are included in the response.

### PR DESCRIPTION
**Pull request by seratch**
###  Summary

This pull request resolves #1397 by generating the web-api response types using the latest source data. To learn the improvement on the Java SDK side, refer to this commit: https://github.com/slackapi/java-slack-sdk/commit/ef33b86486426ea1e6e55ebb44cc0b69d2caa421

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: packages/web-api/src/response/ConversationsListResponse.ts [hunk 1]
```diff
@@ -50,6 +50,9 @@ export interface Channel {
   shared_team_ids?:            string[];
   internal_team_ids?:          string[];
   previous_names?:             string[];
+  user?:                       string;
```
Comment (seratch) [inline packages/web-api/src/response/ConversationsListResponse.ts:53]: These are the ones reported at #1397

File: packages/web-api/src/response/ConversationsOpenResponse.ts [hunk 2]
```diff
@@ -116,11 +115,6 @@ export interface Text {
   verbatim?: boolean;
 }
 
-export interface ElementElementClass {
```
Comment (seratch) [inline packages/web-api/src/response/ConversationsOpenResponse.ts:119]: The changes here mean the redundant types are removed

File: packages/web-api/src/response/SearchAllResponse.ts [hunk 3]
```diff
@@ -490,75 +490,10 @@ export interface Previous {
   text?:        string;
   iid?:         string;
   permalink?:   string;
-  attachments?: MatchAttachment[];
+  attachments?: Attachment[];
   blocks?:      Block[];
 }
 
-export interface Previous2 {
```
Comment (seratch) [inline packages/web-api/src/response/SearchAllResponse.ts:497]: The changes here mean the redundant types are removed

File: packages/web-api/src/response/SearchMessagesResponse.ts [hunk 4]
```diff
@@ -382,75 +382,10 @@ export interface Previous {
   text?:        string;
   iid?:         string;
   permalink?:   string;
-  attachments?: MatchAttachment[];
+  attachments?: Attachment[];
   blocks?:      Block[];
 }
 
-export interface Previous2 {
```
Comment (seratch) [inline packages/web-api/src/response/SearchMessagesResponse.ts:389]: The changes here mean the redundant types are removed

**Review by filmaj — APPROVED** [review]
LGTM, and thanks for sharing the changes on the Java tester side.

**Pull request merged**